pageserver/src/walredo.rs

```diff
@@ -195,26 +156,25 @@ impl WalRedoManager for PostgresRedoManager {
         base_img: Option<Bytes>,
         records: Vec<WALRecord>,
     ) -> Result<Bytes, WalRedoError> {
-        // Create a channel where to receive the response
-        let (tx, rx) = mpsc::channel::<Result<Bytes, WalRedoError>>();
 
         let request = WalRedoRequest {
             rel,
             blknum,
             lsn,
             base_img,
             records,
-            response_channel: tx,
         };
 
-        self.request_tx
-            .lock()
-            .unwrap()
-            .send(request)
-            .expect("could not send WAL redo request");
+        // launch the WAL redo process on first use
+        let mut process_guard = self.process.lock().unwrap();
+        if process_guard.is_none() {
```

Comment:
[Option::get_or_insert_with](https://doc.rust-lang.org/std/option/enum.Option.html#method.get_or_insert_with) might be convenient to use here